### PR TITLE
Add note and example to array_fill() to clarify the effect of a negative start_index in PHP 8

### PR DESCRIPTION
**Pull request by overflowerror**

### CODE REVIEW

File: reference/array/functions/array-fill.xml [hunk 1]
```diff
@@ -33,12 +33,13 @@
       <para>
        The first index of the returned array.
       </para>
-      <para>
-       If <parameter>start_index</parameter> is negative, 
-       the first index of the returned array will be 
-       <parameter>start_index</parameter> and the following 
+      <para> 
+       Up to (excluding) PHP 8, if
```
Comment (cmb69) [inline reference/array/functions/array-fill.xml:37]:
```suggestion
       Prior to PHP 8.0.0, if 
```

File: reference/array/functions/array-fill.xml [hunk 2]
```diff
@@ -108,13 +107,53 @@ Array
...
+    <title>negative <parameter>start_index</parameter> (PHP &lt; 8)</title>
+    <programlisting role="php">
+<![CDATA[
+<?php
+$b = array_fill(-2, 4, 'pear');
+print_r($b);
+?>
+]]>
+    </programlisting>
+    &example.outputs;
```
Comment (cmb69) [inline reference/array/functions/array-fill.xml:123]:
```suggestion
    &example.outputs.8;
```

File: reference/array/functions/array-fill.xml [hunk 3]
```diff
@@ -108,13 +107,53 @@ Array
...
+    <title>negative <parameter>start_index</parameter> (PHP &gt;= 8)</title>
+    <programlisting role="php">
+<![CDATA[
+<?php
+$c = array_fill(-2, 5, 'apple');
+print_r($c);
+?>
+]]>
+    </programlisting>
+    &example.outputs;
```
Comment (cmb69) [inline reference/array/functions/array-fill.xml:146]:
```suggestion
    &example.outputs.8;
```

**Review by cmb69 — COMMENTED** [review]
Thank you for the PR!

File: reference/array/functions/array-fill.xml [hunk 4]
```diff
@@ -108,13 +107,53 @@ Array
...
+    <title>negative <parameter>start_index</parameter> (PHP &lt; 8)</title>
+    <programlisting role="php">
+<![CDATA[
+<?php
+$b = array_fill(-2, 4, 'pear');
+print_r($b);
+?>
+]]>
+    </programlisting>
+    &example.outputs.8;
```
Comment (Crell) [inline reference/array/functions/array-fill.xml:123]: This is for pre-8.0, so I think we need a new `&example.outputs.74` to use here?

Comment (cmb69) [reply reference/array/functions/array-fill.xml:123]: Documentation is still for PHP 7.0+, so `&example.outputs.7;` seems appropriate.

Comment (cmb69) [reply reference/array/functions/array-fill.xml:123]: Anyhow, I don't think we should have two different examples; instead I prefer a single example, which shows the output for PHP 7 and 8.

**Comment by overflowerror** [comment]
#1767 already fixed this.

**Pull request closed without merging**